src/transformers/utils/import_utils.py

```diff
@@ -221,6 +221,10 @@ def _is_package_available(pkg_name: str, return_version: bool = False) -> Union[
 _torch_available = False
 if USE_TORCH in ENV_VARS_TRUE_AND_AUTO_VALUES and USE_TF not in ENV_VARS_TRUE_VALUES:
     _torch_available, _torch_version = _is_package_available("torch", return_version=True)
+    if _torch_available:
+        _torch_available = version.parse(_torch_version) >= version.parse("2.1.0")
+        if not _torch_available:
+            logger.warning(f"Disabling PyTorch because PyTorch >= 2.1 is required but found {_torch_version}")
```

Comment:
@Rocketknight1 Are you also OK to include this new block of logic?

Reply:
I prefer an exception.

Reply:
hmm, since torch is not a hard dependency, after a 2nd though, it might be undesired to change to `_torch_available=False` in an automatically way. But I think this really rarely happens maybe and we can give it a try.

Considering we are not raise values in this file for similar situations, let's go with a warning like here and see how it goes. Thank you @Rocketknight1 !